overlord/servicestate/service_control.go

```diff
@@ -40,6 +40,7 @@ type ServiceAction struct {
 	Action         string   `json:"action"`
 	ActionModifier string   `json:"action-modifier,omitempty"`
 	Services       []string `json:"services,omitempty"`
+	ActionIsOnSnap bool     `json:"action-on-snap"`
```

Comment:
Do we need this flag? This is implied by having empty list of Services, in which case doServiceControl expands to all services and we can special case it for restart there?

Reply:
In the first version of this PR I was indeed clearing the list of Services. But that was causing many tests to fail, and I am afraid that if there's some code path which is not covered by tests, it could cause some bad surprises to our users. Therefore I reverted that part, and now the Services string can be filled even if the action was on the snap -- like it currently is in `master`. So, setting this flag is really the only change in this branch :-)

Reply:
Yes, I did fix all the unit tests before, but then I got three spread tests failing :-(

I spent some time debugging them, but the process was slow and I after a couple of hours I still had no idea about which code-paths I should be looking at. So, since in any case we are expanding the list already, I thought that having a flag would be both more efficient and more safe (with respect of avoiding possible regressions).

Reply:
Failing spread test do not necessarily mean regressions, oftentimes they just have expectations that need tweaking. We should understand what was the cause of the failure. As I said, conceptually having empty service and expanding them in doServiceControl should make not difference (unless I miss something obscure). If you can point me at the commit with your older implementation that was causing this, I'm happy to take a look at these failures and help with understanding them.

Reply:
OK, I restored the old branch here: https://github.com/snapcore/snapd/pull/10292

Still waiting for the spread tests to complete :-)